src/AbstractOperations/computed_field.jl

```diff
@@ -89,3 +89,62 @@ end
 
     @inbounds data[i′, j′, k′] = operand[i′, j′, k′]
 end
+
+struct FusedComputedFields
+    fields::Vector{ComputedField}
+    grid::AbstractGrid
+    datas::Vector{AbstractArray}
+    operands::Vector{AbstractOperation}
+    indices::Tuple
+
+    function FusedComputedFields(fields::Vector{ComputedField})
+        grid = fields[1].grid
+        indices = fields[1].indices
+        arch = architecture(fields[1])
+        sz = size(fields[1])
+        for field in fields[2:end]
+            @assert field.grid === grid
+            @assert field.indices == indices
+            @assert architecture(field) === arch
+            @assert size(field) == sz
```

Comment:
We typically like to give more explicit error messages, so we don't use `@assert` to throw errors for incorrect input. (In other words, its more helpful to the user to say something like "one of your grids is different"). There's also the concern that it can be disabled

https://docs.julialang.org/en/v1/base/base/#Base.@assert

Reply:
The new commit throws `ArgumentError` instead of using `@assert`.